app/redux/shared_actions/SetsActionCreators.js

```diff
@@ -66,11 +68,53 @@ export const saveHistorySet = (setID, exercise = null, weight = null, metric = n
     return action;
 };
 
-export const endSet = () => (dispatch, getState) => {
+export const endSet = (was_sanity_check=false, duration=0, manually_started=false) => (dispatch, getState) => {
```

Comment:
Why is duration passed in? Should pull the end set timer from the store

Reply:
Killed